File: Euclid/Messages/Incoming/UPDATE.cs
using Euclid.Game;
using Euclid.Network.Streams;
using Euclid.Storage.Database.Access;

namespace Euclid.Messages.Incoming
{
    class UPDATE : IMessageEvent
    {
        public void Handle(Player player, Request request)
        {
            var registerValues = request.GetKeyValues();

            if (!registerValues.ContainsKey("name") ||
                !registerValues.ContainsKey("password") ||
                !registerValues.ContainsKey("email") ||
                !registerValues.ContainsKey("figure") ||
                !registerValues.ContainsKey("directMail") ||
                !registerValues.ContainsKey("birthday") ||
                !registerValues.ContainsKey("phonenumber") ||
                !registerValues.ContainsKey("customData") ||
                !registerValues.ContainsKey("has_read_agreement") ||
                !registerValues.ContainsKey("sex") ||
                !registerValues.ContainsKey("country"))
            {
                return;
            }

            var data = player.Details;

            data.Name = registerValues["name"];
            data.Password = registerValues["password"];
            data.Email = registerValues["email"];
            data.Figure = registerValues["figure"];
            data.DirectMail = registerValues["directMail"] == "1";
            data.Birthday = registerValues["birthday"];
            data.PhoneNumber = registerValues["phonenumber"];
            data.CustomData = registerValues["customData"];
            data.Sex = registerValues["sex"] == "Male" ? "M" : "F";
            data.Country = registerValues["country"];

            UserDao.SaveOrUpdate(data);
        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/UPDATE.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.messages.IMessageEvent;
import euclid.network.streams.Request;
import euclid.storage.database.access.UserDao;

public class UPDATE implements IMessageEvent {

    @Override
    public void handle(Player player, Request request) {
        var registerValues = request.getKeyValues();

        if (!registerValues.containsKey("name") ||
                !registerValues.containsKey("password") ||
                !registerValues.containsKey("email") ||
                !registerValues.containsKey("figure") ||
                !registerValues.containsKey("directMail") ||
                !registerValues.containsKey("birthday") ||
                !registerValues.containsKey("phonenumber") ||
                !registerValues.containsKey("customData") ||
                !registerValues.containsKey("has_read_agreement") ||
                !registerValues.containsKey("sex") ||
                !registerValues.containsKey("country")) {
            return;
        }

        var data = player.getDetails();

        data.setName(registerValues.get("name"));
        data.setPassword(registerValues.get("password"));
        data.setEmail(registerValues.get("email"));
        data.setFigure(registerValues.get("figure"));
        data.setDirectMail("1".equals(registerValues.get("directMail")));
        data.setBirthday(registerValues.get("birthday"));
        data.setPhoneNumber(registerValues.get("phonenumber"));
        data.setCustomData(registerValues.get("customData"));
        data.setSex("Male".equals(registerValues.get("sex")) ? "M" : "F");
        data.setCountry(registerValues.get("country"));

        UserDao.saveOrUpdate(data);
    }
}
